src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,169 @@
+        });
+    }
+
+    @Test
+    void register_loginIsNull_notOk() {
+        userSecond.setLogin(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userSecond);
+        });
+    }
+
+    @Test
+    void register_loginExists_notOk() {
+        userSecond.setLogin(MODEL_FIRST_USER_LOGIN);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userSecond);
+        });
+    }
+
+    @Test
+    void register_loginIsLessThanSixChars_notOk() {
+        userSecond.setLogin("");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userSecond);
+        });
+        userSecond.setLogin("Ab/@1");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userSecond);
+        });
+    }
+
+    @Test
+    void register_loginIsSixCharsAndMore_ok() {
+        userSecond.setLogin("Den$12");
+        registrationService.register(userSecond);
+        assertSame(userSecond, storageDao.get(userSecond.getLogin()));
+        userSecond.setLogin("Alice1!");
+        assertSame(userSecond, storageDao.get(userSecond.getLogin()));
+        userSecond.setLogin("$1Bohdan@2");
+        assertSame(userSecond, storageDao.get(userSecond.getLogin()));
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        userSecond.setPassword(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userSecond);
+        });
+    }
+
+    @Test
+    void register_passwordIsLessThanSixChars_notOk() {
+        userSecond.setPassword("b");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userSecond);
+        });
+        userSecond.setPassword("Bob32");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userSecond);
+        });
+    }
+
+    @Test
+    void register_passwordIsSixCharsAndMore_ok() {
+        userSecond.setPassword("123456");
+        registrationService.register(userSecond);
+        assertSame(userSecond, storageDao.get(userSecond.getLogin()));
+        userSecond.setPassword("12//56.(");
+        assertSame(userSecond, storageDao.get(userSecond.getLogin()));
+        userSecond.setPassword("Bob@$.3456");
+        assertSame(userSecond, storageDao.get(userSecond.getLogin()));
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        userSecond.setAge(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userSecond);
+        });
+    }
+
+    @Test
+    void register_ageIsNegative_notOk() {
+        userSecond.setAge(-18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userSecond);
+        });
+    }
+
+    @Test
+    void register_ageIsLessThan18_notOk() {
```

Comment:
Do not hardcode values in name
```suggestion
    void register_lowAge_notOk() {
```